packages/cursorless-engine/src/languages/TreeSitterQuery/queryPredicateOperators.ts

```diff
@@ -97,6 +97,15 @@ class ChildRange extends QueryPredicateOperator<ChildRange> {
  * A predicate operator that modifies the range of the match to shrink to regex
  * match.  For example, `(#shrink-to-match! @foo "\\S+")` will modify the range
  * of the `@foo` capture to exclude whitespace.
+ *
+ * If convenient, you can use a special capture group called `keep` to indicate
+ * the part of the match that should be kept.  For example,
+ *
+ * ```
+ * (#shrink-to-match! @foo "^\s+(?<keep>.*)$")
```

Comment:
Why not just use the first group? <keep> just feels like unnecessary code.

Reply:
I just worry that you'll use a `()` for another purpose and be surprised when all of a sudden it narrows to that. I wanted to make sure it was fully intentional

Reply:
Not a fan of non capture groups? You want it more explicit?

Reply:
Yeah I know non-capture groups exist, but my habit is to just use `()` without thinking. Idk maybe that's a bad habit? I don't feel super strongly, just trying to reduce surprise

Reply:
Heh maybe. A named group is fine though :)